x-pack/plugins/ml/public/util/index_utils.js

```diff
@@ -71,7 +71,7 @@ export function timeBasedIndexCheck(indexPattern, showNotification = false) {
     if (showNotification) {
       const message = `The index pattern ${indexPattern.title} is not time series based. \
         Anomaly detection can only be run over indices which are time based.`;
-      notify.warning(message, { lifetime: 0 });
+      toastNotifications.addWarning(message);
     }
```

Comment:
The index pattern ${indexPattern.title} is not based on a time series.

Anomaly detection only runs over time-based indices.